#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/toys/protos/RubberDuck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/balls/protos/FifaSoccerBall.proto"
EXTERNPROTO "../protos/MobilePhone.proto"
EXTERNPROTO "../protos/Wallet.proto"
EXTERNPROTO "../protos/Keys.proto"
EXTERNPROTO "../protos/Jewelry.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/apartment_structure/protos/Wall.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.01999861481467208 -0.0022777631633297364 0.9997974130795029 2.9148235176102992
  position 2.8265380998833476 1.0012243745775475 0.05850208771366916
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 10 10
}
Solid {
  translation 2 0.9830260732449472 0.049215133509278294
  rotation -1 -4.4147782458756385e-17 3.552945215224071e-17 1.5707951778241853
  children [
    Transform {
      scale 0.03 0.03 0.03
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Mesh {
            url [
              "../objects_webots/obj/keys/keys.obj"
            ]
          }
        }
      ]
    }
  ]
  name "Keys"
  boundingObject Box {
    size 0.2 0.016 0.1
  }
  physics Physics {
  }
  recognitionColors [
    0.6 0.3 0.1
  ]
}
Solid {
  translation 0.006357679141904482 2 0.059057162904858754
  rotation -4.1875889134178733e-17 1 -3.731494742137675e-17 3.141592652619402
  children [
    Transform {
      scale 0.1 0.1 0.1
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Mesh {
            url [
              "C:/Users/Keerthy/Downloads/objects_webots/obj/mobile/scene.obj"
            ]
          }
        }
      ]
    }
  ]
  name "Smartphone"
  boundingObject Box {
    size 0.1 0.2 0.12
  }
  physics Physics {
  }
  recognitionColors [
    0.6 0.3 0.1
  ]
}
Wall {
  translation 0 -2 0
  rotation 0 0 1 0.7853996938995746
  size 0.2 3 2.4
}
Wall {
  translation 0 -3 0
  rotation 0 0 1 1.5708003061004252
  name "wall(1)"
  size 0.2 8 2.4
}
Wall {
  translation -2.6 3 0
  rotation 0 0 1 -0.5236003061004253
  name "wall(2)"
  size 0.2 2 2.4
}
Wall {
  translation -4 1 0
  name "wall(3)"
  size 0.2 5 2.4
}
Wall {
  translation 2 3 0
  name "wall(4)"
  size 0.2 2 2.4
}
Wall {
  translation 0 4 0
  rotation 0 0 1 1.5708003061004252
  name "wall(5)"
  size 0.2 8 2.4
}
Wall {
  translation 4 0 0
  name "wall(6)"
  size 0.2 9 2.4
}
Jewelry {
  translation -1 -2 0.009971747198964988
}
Keys {
  translation 1 2 0.03997781030168174
}
Wallet {
  translation -1 1 0.00745879799876944
}
MobilePhone {
  translation 2 -2 0.004961544798997365
}
DEF rover Robot {
  translation 0.2670016165813276 -0.26700103013569637 0.09295738140993215
  rotation 0.0014629329146273273 0.002179814751381451 -0.9999965541115314 0.7854018787875894
  children [
    Camera {
      translation 0.29 -0.1 0.1
      children [
        DEF cambox Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "CAM"
      boundingObject USE cambox
      physics Physics {
      }
      width 640
      height 480
      recognition Recognition {
      }
    }
    DEF ds_left DistanceSensor {
      translation 0.36 0.1 0
      rotation 0 0 1 0.3927
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 1 0.196078 0.0901961
          }
          geometry Box {
            size 0.025 0.025 0.025
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.4 1000 0
      ]
      numberOfRays 2
      resolution 1024
    }
    DEF ds_right DistanceSensor {
      translation 0.36 -0.1 0
      rotation 0 0 1 -0.3927
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 1 0.196078 0.0901961
          }
          geometry Box {
            size 0.025 0.025 0.025
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.4 1000 0
      ]
      numberOfRays 2
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 0.3 0.3 0.3
        metalness 0
      }
      geometry Box {
        size 0.7 0.4 0.1
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 16.606573418571415
        axis 0 1 0
        anchor 0.175 0.225 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.175 0.225 0
        rotation -0.3227223891399547 0.6692709627371596 -0.6692732162463064 3.7659316868157653
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.6 0.552941 0.47451
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.1
            }
          }
        ]
        boundingObject USE wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 16.610586778698494
        axis 0 1 0
        anchor 0.175 -0.225 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.175 -0.225 0
        rotation -0.3242055270511538 0.6689124617760657 -0.6689147140780857 3.7686169934690037
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.6 0.552941 0.47451
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.1
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 16.60194496823091
        axis 0 1 0
        anchor -0.175 0.225 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.175 0.225 0
        rotation -0.3210125569512449 0.6696819835384504 -0.669684238431539 3.762833047205447
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.6 0.552941 0.47451
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.1
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 16.60657738863443
        axis 0 1 0
        anchor -0.175 -0.225 0
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.175 -0.225 0
        rotation -0.3227238560335716 0.6692706090694904 -0.6692728625774215 3.765934343862972
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.6 0.552941 0.47451
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.1
            }
          }
        ]
        name "solid(3)"
        boundingObject USE wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    DEF LIDAR_support Shape {
      appearance PBRAppearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.05
      }
    }
    Lidar {
      translation 0 0 0.24
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.01
            radius 0.05
          }
        }
      ]
      name "LIDAR"
      numberOfLayers 5
    }
    DEF IMU InertialUnit {
      translation -0.2 0 0.1
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "IMU"
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "obj_nav"
}
RubberDuck {
  translation 3.0000000000001443 2 0.05988432768754436
  rotation -8.557662089987826e-08 -0.9999999999999963 -1.1648292809403895e-11 8.855139504049391e-07
  scale 3
}
FifaSoccerBall {
  translation -3.6 -1 0.11182507199993243
}
